python/lsst/analysis/tools/tasks/associatedSourcesTractAnalysis.py

```diff
@@ -123,6 +188,77 @@ def prepareAssociatedSources(cls, skymap, tract, sourceCatalogs, associatedSourc
 
         return dataFiltered
 
+    def applyAstrometricCorrections(self, dataJoined, astrometricCorrectionCatalogs, visitTable):
+        """Use proper motion/parallax catalogs to shift positions to median
+        epoch of the visits.
+
+        Parameters
+        ----------
+        dataJoined : `astropy.table.Table`
+            Table containing source positions, which will be modified in place.
+        astrometricCorrectionCatalogs: `dict` [`pd.DataFrame`]
+            Dictionary keyed by band with proper motion and parallax catalogs.
+        visitTable : `pd.DataFrame`
+            Table containing the MJDs of the visits.
+        """
+        for band in np.unique(dataJoined["band"]):
+            bandInd = dataJoined["band"] == band
+            bandSources = dataJoined[bandInd]
+            # Add key for sorting below.
+            bandSources["__index__"] = np.arange(len(bandSources))
+            bandSourcesDf = bandSources.to_pandas()
+            meanRAs = bandSourcesDf.groupby("obj_index")["coord_ra"].aggregate("mean")
+            meanDecs = bandSourcesDf.groupby("obj_index")["coord_dec"].aggregate("mean")
+
+            bandPMs = astrometricCorrectionCatalogs[band]
+            with Matcher(meanRAs, meanDecs) as m:
```

Comment:
Having just started looking at `smatch`, I'm wondering why you're using the context manager here, instead of just calling `smatch.match`? You're only ever doing the match once with a given set of coordinates, so you're not gaining anything from creating a `Matcher` instance.

Reply:
I just asked Eli about this, because I've just been copying the pattern I usually see in the stack. He says there's a memory-intensive KD tree being built by `Matcher`, so it either needs to be in a context manager or explicitly deleted.